### PR TITLE
Fix problem with path separators

### LINKED ISSUE #906: Different path separators do not filter autogenerated files
GeneratedFileFilter does not work when the file paths in the .pb file have different path separator than the files reported by the plugin.

### PR DESCRIPTION
**Pull request by valhristov**
Fix #906

### CODE REVIEW

File: sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/GeneratedFileFilter.java [hunk 1]
```diff
@@ -39,7 +42,7 @@
   private final ProtobufImporters protobufImporters;
 
   private boolean initialized;
-  private Set<String> generatedFilePaths;
+  private Set<Path> generatedFilePaths = new HashSet<>();
```
Comment (valhristov) [inline sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/GeneratedFileFilter.java]: The change from Path to String broke the filter, I am returning Path back as the recommended way to work with paths.

File: sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/GeneratedFileFilter.java [hunk 2]
```diff
@@ -26,8 +26,11 @@
 import org.sonarsource.dotnet.shared.plugins.protobuf.FileMetadataImporter;
 import org.sonarsource.dotnet.shared.plugins.protobuf.ProtobufImporters;
 
+import javax.swing.text.StyledEditorKit;
 import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.HashSet;
+import java.util.Map;
```
Comment (SonarTech) [inline sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/GeneratedFileFilter.java]: MINOR Remove this unused import 'java.util.Map'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

File: sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/GeneratedFileFilter.java [hunk 3]
```diff
@@ -26,8 +26,11 @@
 import org.sonarsource.dotnet.shared.plugins.protobuf.FileMetadataImporter;
 import org.sonarsource.dotnet.shared.plugins.protobuf.ProtobufImporters;
 
+import javax.swing.text.StyledEditorKit;
```
Comment (SonarTech) [inline sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/GeneratedFileFilter.java]: MINOR Remove this unused import 'javax.swing.text.StyledEditorKit'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

File: sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/GeneratedFileFilter.java [hunk 4]
```diff
@@ -26,8 +26,11 @@
 import org.sonarsource.dotnet.shared.plugins.protobuf.FileMetadataImporter;
 import org.sonarsource.dotnet.shared.plugins.protobuf.ProtobufImporters;
 
+import javax.swing.text.StyledEditorKit;
 import java.nio.file.Path;
+import java.nio.file.Paths;
```
Comment (SonarTech) [inline sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/GeneratedFileFilter.java]: MINOR Remove this unused import 'java.nio.file.Paths'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/GeneratedFileFilter.java [hunk 5]
```diff
@@ -54,9 +57,9 @@ public GeneratedFileFilter(AbstractConfiguration config) {
   public boolean accept(InputFile inputFile) {
     initOnce();
 
-    boolean isGenerated = generatedFilePaths.contains(inputFile.absolutePath());
+    boolean isGenerated = generatedFilePaths.contains(inputFile.path());
```
Comment (valhristov) [inline sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/GeneratedFileFilter.java]: FYI `absolutePath()` returns `String`, `path()` returns `Path`

**Pull request merged**